.github/scripts/check-stale-prs.js

```diff
@@ -0,0 +1,195 @@
+import { Octokit } from '@octokit/rest';
+import dayjs from 'dayjs';
+
+// ----- Configuration -----
+const REPO_OWNER = 'konflux-ci';
+const REPO_NAME = 'konflux-ui';
+const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
+const SLACK_WEBHOOK_URL = process.env.SLACK_WEBHOOK_URL;
+const TEST_MODE = process.env.TEST_MODE === 'true';
+
+if (!GITHUB_TOKEN) throw new Error('GITHUB_TOKEN is not set');
+if (!TEST_MODE && !SLACK_WEBHOOK_URL) throw new Error('SLACK_WEBHOOK_URL is not set');
+
+const octokit = new Octokit({ auth: GITHUB_TOKEN });
+
+// ----- Helpers -----
+const getPRs = (params = {}) =>
+  octokit.paginate(octokit.pulls.list, {
+    owner: REPO_OWNER,
+    repo: REPO_NAME,
+    state: 'open',
+    per_page: 100,
+    ...params,
+  });
+
+const getComments = (prNumber) =>
+  octokit.paginate(octokit.issues.listComments, {
+    owner: REPO_OWNER,
+    repo: REPO_NAME,
+    issue_number: prNumber,
+    per_page: 100,
+  });
+
+const getReviews = (prNumber) =>
+  octokit.paginate(octokit.pulls.listReviews, {
+    owner: REPO_OWNER,
+    repo: REPO_NAME,
+    pull_number: prNumber,
+    per_page: 100,
+  });
+
+// ----- Main logic -----
+const checkStalePRs = async () => {
+  const prs = await getPRs();
+  const draftPRs = [];
+  const noCommentsIn2Days = [];
+  const approvedButOpen = [];
+  const openOver2Weeks = [];
+  const openOver1Month = [];
+  const openOver2Months = [];
+
+  const now = dayjs();
+
+  for (const pr of prs) {
+    const [comments, reviews] = await Promise.all([getComments(pr.number), getReviews(pr.number)]);
+    const daysOpen = now.diff(dayjs(pr.created_at), 'day');
+    const prInfo = { number: pr.number };
+
+    if (pr.draft) {
+      draftPRs.push(prInfo);
+      continue;
+    }
+
+    // Track latest activity
+    let latestActivity = null;
+    const latestReviewByUser = new Map();
+
+    comments.forEach((c) => {
+      const ts = dayjs(c.updated_at);
+      if (!latestActivity || ts.isAfter(latestActivity)) latestActivity = ts;
+    });
+
+    reviews.forEach((r) => {
+      if (!r.submitted_at) return;
+      const ts = dayjs(r.submitted_at);
+      if (!latestActivity || ts.isAfter(latestActivity)) latestActivity = ts;
+
+      const login = r.user?.login;
+      if (!login) return;
+
+      const prior = latestReviewByUser.get(login);
+      if (!prior || ts.isAfter(prior.submitted_at)) {
+        latestReviewByUser.set(login, { state: r.state, submitted_at: ts });
+      }
+    });
+
+    const activeApprovals = Array.from(latestReviewByUser.values()).filter(
+      (rev) => rev.state === 'APPROVED',
+    );
+
+    if (activeApprovals.length >= 2 && !pr.merged_at) approvedButOpen.push(prInfo);
+
+    const inactiveForTwoDays = latestActivity
```

Comment:
This will return all PRs after the weekend, right? Is it intended? We may consider excluding weekends by, for example, adding a check like "if it's Monday today, check for 4 days without review" to include 2 previous working days. WDYT?

Reply:
updatred.